### PR TITLE
Add onboarding support for new countries in WCPay

### LINKED ISSUE #36904: [Enhancement]: Add new countries to WCPay onboarding
### Describe the solution you'd like

Part of launching new countries support in WCPay, add these countries as part of the onboarding:

- Denmark
- Greece
- Norway
- Slovakia
- Finland
- Estonia
- Lithuania
- Slovenia
- Latvia
- Cyprus
- Malta
- Luxembourg

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by elazzabi**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #36904.

This PR adds onboarding support for 12 new countries in WCPay:

Before this PR:
```
[ 'US', 'PR', 'AU', 'CA', 'DE', 'ES', 'FR', 'GB', 'IE', 'IT', 'NZ', 'AT', 'BE', 'NL', 'PL', 'PT', 'CH', 'HK', 'SG' ].length
19 
```

After this PR:
```
[ 'US', 'PR', 'AU', 'CA', 'CY', 'DE', 'DK', 'EE', 'ES', 'FI', 'FR', 'GB', 'GR', 'IE', 'IT', 'LU', 'LT', 'LV', 'NO', 'NZ', 'MT', 'AT', 'BE', 'NL', 'PL', 'PT', 'CH', 'HK', 'SI', 'SK', 'SG' ].length
31 
```

**This PR targets the next version of WC `7.6.0`**

WCPay PR for info: https://github.com/Automattic/woocommerce-payments/pull/5536

- [x] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### Testing instructions:

To trigger the OBW fallback, disable marketplace suggestions:  
Settings > [WooCommere.com](http://woocommere.com/) > untick Display suggestions within WooCommerce

1. Get this PR locally.
2. Change your store location to one of the countries in this list (eg. Denmark, Finland, etc.).
3. Trigger the WooCommerce onboarding wizard.
4. Make sure WC Payments is showing as the suggested payment gateway in the "Set up payments" step.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by barryhughes** [comment]
Hi @elazzabi! 

Could we [expand the testing instructions](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions) just a little? I reviewed [WC Pay PR 5536](https://github.com/Automattic/woocommerce-payments/pull/5536) and though this is probably the proverbial 'silly question' I'm still just a little unsure if I'd expect to see a recommendation during onboarding (ie, when setting up WooCommerce for the first time), but before having installed the WC Payments plugin?

Is the new behavior only realized when this branch **and** the branch in that other PR are both active?

**Comment by elazzabi** [comment]
Hello @barryhughes, thank you for taking a look! And sorry for the late reply, I was AFK the past few days.

For your question, yes there should be a recommendation during the onboarding for WooCommerce **without** the WC Payments being installed (it shouldn't be a requirement). Something like this:

Screenshot 2023-03-06 at 22 00 39

Given I'm not familiar with the code base, I followed [a similar PR](https://github.com/woocommerce/woocommerce-admin/pull/7558/files) that was introduced a while ago, thinking this would be an easy task, and forgot to add testing instructions. Sorry about that!

Now, with testing, I'm having some problems trying this locally for some reason. The file changes arent synced to my local dev instance (the one that gets created after running `npm run env:start`), so I'm not able to test those changes.

I removed all transients, and I also tried to remove countries already supported to see if they'll break, but it looks like it's not working either. This looks like a caching issue or something... I'll keep looking, but if you have any pointers, that would be perfect!

**Comment by barryhughes** [comment]
Thanks for the screenshot—that is exactly the sort of thing I was trying to find but couldn't find on my first pass (not sure if I missed it, or just didn't create the necessary conditions). I'll circle back and give this a second spin a little later in the week :+1:

**Comment by barryhughes** [comment]
Hi, @elazzabi!

- The change itself looks *safe,* I don't think it is going to introduce any problems.
- That said, I'm unable to say if it's doing what it should (see further notes) and you might find others also require more detailed testing instructions at a later stage in the release cycle.

If I understand things correctly, Denmark (DK) is being added to the list of countries for which WC Payments will be recommended. However, setting my base country to Denmark and then navigating to the **WooCommerce ▸ Settings ▸ Payments** screen isn't enough to see the recommendation (is that sufficient to test this?). I did dig a little further, and noticed:

- Although `DK` has been added via your changes [here](https://github.com/woocommerce/woocommerce/pull/36906/files#diff-6e8b1c397d6ca6f86863f70ea1fe36b8ede207b97fb40265e6a739f230d13536R77) and [here](https://github.com/woocommerce/woocommerce/pull/36906/files#diff-d55ecc7aa69536540731334e4cd0e4647485d1130cfb2c987120f1d75f247298R436), when the time comes for the spec for `'woocommerce_payments'` to be [evaluated over here](https://github.com/woocommerce/woocommerce/blob/7.4.1/plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php#L43), those new countries are missing from the list.
- It seems like they are [cached in a transient](https://github.com/woocommerce/woocommerce/blob/7.4.1/plugins/woocommerce/src/Admin/DataSourcePoller.php#L108) and, if I clear that, they are ultimately [sourced from a remote location](https://github.com/woocommerce/woocommerce/blob/7.4.1/plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/PaymentGatewaySuggestionsDataSourcePoller.php#L20-L22) (which doesn't yet seem to return the new countries).

tl;dr — I don't see any safety problems with the change, but am also not sure if it is working as expected. This leaves me somewhat hesitant to approve+merge. Happy to discuss further, though, as I'm unfamiliar with this part of the codebase and may simply be misunderstanding how things are meant to work.

**Comment by barryhughes** [comment]
...Building on my last comment, and looking at the history of this code, I actually wonder if it would make more sense to request a review from Mothra?

**Comment by elazzabi** [comment]
Hello @barryhughes 👋 

Thank you for testing this. I updated the PR with specific test steps after figuring this out myself. Sorry for the confusion 🙇 

As you said, there are two parts: reading from local changes and from the remote source (in addition to caching the result in a transient). By default, we read from the remote source before falling back to the local JSON.

So, in order to test this, we need to remove the cache, and then change the remote source to a random URL to trigger an error => the local file will kick in. I updated the PR description with this new information. Also, I included the PR where we'll change the remote location too.

Also, thank you for the suggestion. I asked for a review from Mothra 🙇

**Review by mattsherman — COMMENTED** [review]
I tested this and was unable to get WCPay to show up when using the local fallback data. This confused me and I spent a lot of time digging into this...

I believe the root of the reason why this doesn't work is that:

1. If WCPay is supported for the store's country, then the payment recommendations are *not* shown below the list of payment methods ([source](https://github.com/woocommerce/woocommerce/blob/3809392f7c620bd7035c1d5f93e63e46ccfb4476/plugins/woocommerce-admin/client/payments/payment-recommendations.tsx#L87))
2. If WCPay is supported for the store's country, WCPay should show up *directly* in the payment methods list as a promotion (as shown in the screenshot in the comments of this PR). However...
3. The WCPay promotion row only shows up if the promotion is valid for the store's country. To determine this, a remote API call is made to get the list of promotions ([source](https://github.com/woocommerce/woocommerce/blob/3809392f7c620bd7035c1d5f93e63e46ccfb4476/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php#L66)). However, there is *no local fallback* for this data ([source](https://github.com/woocommerce/woocommerce/blob/3809392f7c620bd7035c1d5f93e63e46ccfb4476/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php#L154)), so the promotion only shows up if the remote API data has WCPay marked as being supported (as will be added in 16269-gh-Automattic/woocommerce.com). (See the similar [payment gateway suggestions source](https://github.com/woocommerce/woocommerce/blob/3809392f7c620bd7035c1d5f93e63e46ccfb4476/plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php#L68) to see how a local fallback was implemented).

So, unless a local fallback is implemented, this will only work once the remote data spec is live.

I'm not sure if this was implemented this way on purpose (not having a local fallback), or if this is a bug. Perhaps @pmcpinto will know more ([original issue where this was added](https://github.com/woocommerce/woocommerce-admin/issues/7319)).

**Comment by adrianduffell** [comment]
I'm also not able to make it appear in the Settings "Recommended payment providers". @elazzabi does it still work for you?

You can also add the new countries to [DefaultFreeExtensions.php](https://github.com/woocommerce/woocommerce/blob/16c7c584627b21534df8630d6c4dcef6e7471214/plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php#L188) to surface the recommendation in the Onboarding Wizard (when its remote API is unavailable).

**Comment by anu-rock** [comment]
As the primary concern of this PR is to ensure WCPay is offered as a suggestion in Woo's onboarding wizard, I've pushed [new changes](https://github.com/woocommerce/woocommerce/pull/36906/commits/021ba5636cd158eaf4ba8cf244f03f650306bb48) as suggested by @adrianduffell and updated the testing instructions. If we can test the onboarding flow successfully, I think the PR will have achieved its objective.

**Comment by anu-rock** [comment]
Also noting that we'll separately work on getting 16269-gh-Automattic/woocommerce.com reviewed and merged (after this PR) since we don't yet support a local fallback for WCPay promotion. So, nothing left to test beside the onboarding wizard for now.

**Review by adrianduffell — APPROVED** [review]
Thanks, it tested well for me in the onboarding wizard!

I'd agree on keeping the other country config in place. It's seems like the fallback problems need to be fixed outside of this PR.
